### DISCUSSION TITLE
Download failed

### DISCUSSION THREAD

**Discussion by avanishb** [aclap-dev/video-downloadhelper#674] [Help and Q&A]
### Your question

{
  "version": "9.0.1.22",
  "target": "google",
  "channel": "stable",
  "page_url": "https://www.jeyadra.com/s/courses/660d465b0fa02d2c71eb7aea/take?loginMsg=1",
  "error": {
    "details": "SideDownload error: : [hls @ 0000021d5815a500] Error when loading first segment 'https://dnqe0v482iyb1.cloudfront.net/spees/w/o/61af88470cf272041d7a8088/v/6651dcde71bdd909cdc82b12/u/664b9498c4298e7b232478e8/t/7faf2533f00f3eb27e2d534abcded4a1/p/assets/videos/61af88470cf272041d7a8088/2024/05/25/6651dcde71bdd909cdc82b12/hls_2M_000.ts'\r\n[in#0 @ 0000021d56998d00] Error opening input: Invalid data found when processing input\r\nError opening input file https://dnqe0v482iyb1.cloudfront.net/spees/w/o/61af88470cf272041d7a8088/v/6651dcde71bdd909cdc82b12/u/664b9498c4298e7b232478e8/t/7faf2533f00f3eb27e2d534abcded4a1/p/assets/videos/61af88470cf272041d7a8088/2024/05/25/6651dcde71bdd909cdc82b12/hls_2M_.m3u8.\r\nError opening input files: Invalid data found when processing input\r\n",
    "downloadable_id": "downloadable_2426704997093365",
    "error": "coapp_failure",
    "id": "coapp_failure_a55239e6-daff-4519-9a93-ef7b743eee80"
  }
}

this is the bug details, I am unable to download a protected video from one of the site. Someone please help me to download it. its a protected video, an online class, have to put password and email. pleaase guide me download it

### Addon & CoApp version and details

version: 9.0.1.22
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "used_sidebar_button": true,
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 4
}

**Comment by Mattmjs** [comment]
It looks protected as you say: #EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29
Doesn't look like it will work.

**Reply by avanishb** [reply]
ok thanks, @Mattmjs any other alternative?

**Reply by Mattmjs** [reply]
This is not just download issue for VDH but any other downloading software. Another tool I use is also stumped.
Note the part in bold.

[generic] hls_2M_: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] hls_2M_: Downloading m3u8 information
[generic] hls_2M_: Checking m3u8 live status
[info] hls_2M_: Downloading 1 format(s): 0
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 9659
[download] Destination: hls_2M_ [hls_2M_].mp4
[download]   0.0% of ~   3.26GiB at    1.01KiB/s ETA Unknown (frag 1/9659)**ERROR: Incorrect AES key length (464 bytes)**

**Reply by avanishb** [reply]
@Mattmjs thank you so much for your effort, if I give you the credentials is it possible to download the video?

**Reply by Mattmjs** [reply]
I don't need your credentials, I've already tried downloading your video based on the info you've provided here.
It's simply not going to work, they have put some protection on it and that means downloading is hitting a road block.
Like a dead end road. If your really needing this video you should investigate screen recording software.
You basically record the video as it plays.
